design/0008-community-plugins.md

```diff
@@ -0,0 +1,48 @@
+# Community Plugins
+
+```
+Author: Yudai Kiyofuji <yudai.kiyofuji@emqx.io>
+Status: Draft
+First Created: 2021-02-21
+EMQ X Version: 4.3
+```
+
+## Abstract
+
+This proposal suggests ways to encourage plugin development by community. 
+
+## Motivation
+
+Allowing people to develop their own plugins is a good way for EMQX to gain popularity.
+To achieve this, it is nice to `approve` some community plugins and let people use them.
+
+## Design
+
+In rebar.config.erl, we can add a list of approved community plugins.
+
+```erlang
+community_deps() ->
+    [
+      {foo_plugin, {git, "https://github.com/###"}}
+    , {bar_plugin, {git, "https://github.com/###"}} 
+    ].
+```
+
+And when a user would like to use one of them, 
+he/she can do so by setting env variable `EMQX_COMMUNITY_PLUGINS=foo_plugin`.
+
+
+ ## Configuration Changes
+
+
+
+ ## Backwards Compatibility
+
+
+ ## Document Changes
+
+
+
+ ## Testing Suggestions
```

Comment:
should describe how a third-party plugin can be tested.
ideally the standalone repo should test itself without having to be injected into the umbrella project.
but if there is no way to avoid that, we may need to find a easy way to test (local and in CI) their app.

e.g. put the app under development in emqx's `_checkouts` dir and use emqx's `make ct --dir=...` to test it.

Reply:
> e.g. put the app under development in emqx's _checkouts dir and use emqx's make ct --dir=... to test it.

You expect these steps are done manually? Or there should be some command to pull the repository and place it in a good place to make test works?

Reply:
How about having `make include https://some-plugin.git` command, which will download `some-plugin` to `apps` directory.

Reply:
It's always tricky having to deal with inverse dependency in tests.
ideally, we should provide a `emqx-mock` or something for standalone repo's eunit and ct.
However I realise that could be another EIP.
So as the first step, we can document it as a manual step.
Nothing is stopping anyone from scripting it though.

Reply:
updated the doc